src/substrait/textplan/converter/PlanPrinterVisitor.cpp

```diff
@@ -7,58 +7,567 @@
+      break;
+    case ::substrait::proto::Expression::Literal::kI16:
+      text << literal.i16() << "_i16";
+      break;
+    case ::substrait::proto::Expression::Literal::kI32:
+      text << literal.i32() << "_i32";
+      break;
+    case ::substrait::proto::Expression::Literal::kI64:
+      text << literal.i64() << "_i64";
+      break;
+    case ::substrait::proto::Expression::Literal::kFp32:
+      text << literal.fp32() << "_fp32";
+      break;
+    case ::substrait::proto::Expression::Literal::kFp64:
+      text << literal.fp64() << "_fp64";
+      break;
+    case ::substrait::proto::Expression::Literal ::kDate: {
+      // TODO -- Format this as a date instead of a delta since an epoch.
+      if (literal.date() >= 0) {
+        text << "\"epoch+" << literal.date() << " days\"_date";
+      } else {
+        text << "\"epoch" << literal.date() << " days\"_date";
+      }
+      break;
+    }
+    case ::substrait::proto::Expression::Literal::kString:
+      text << "\"" << stringEscape(literal.string()) << "\"_string";
+      break;
+    case ::substrait::proto::Expression_Literal::kBinary:
+    case ::substrait::proto::Expression_Literal::kTimestamp:
+    case ::substrait::proto::Expression_Literal::kTime:
+      error_listener_->addError(
+          "Literal of this type not yet supported: " +
+          literal.ShortDebugString());
+      return std::string("UNSUPPORTED_LITERAL_TYPE");
+    case ::substrait::proto::Expression_Literal::kIntervalYearToMonth: {
+      text << "{";
+      bool hasPreviousText = false;
+      if (literal.interval_year_to_month().years() != 0) {
+        text << literal.interval_year_to_month().years() << "years";
+        hasPreviousText = true;
+      }
+      if (literal.interval_year_to_month().months() != 0) {
+        if (hasPreviousText)
+          text << ", ";
+        text << literal.interval_year_to_month().months() << "months";
+      }
+      text << "}_interval_year"; // TODO - Change spec to better name.
+      break;
+    }
+    case ::substrait::proto::Expression_Literal::kIntervalDayToSecond: {
+      text << "{";
+      bool hasPreviousText = false;
+      if (literal.interval_day_to_second().days() != 0) {
+        text << literal.interval_day_to_second().days() << "days";
+        hasPreviousText = true;
+      }
+      if (literal.interval_day_to_second().seconds() != 0) {
+        if (hasPreviousText)
+          text << ", ";
+        text << literal.interval_day_to_second().days() << "seconds";
+        hasPreviousText = true;
+      }
+      if (literal.interval_day_to_second().microseconds() != 0) {
+        if (hasPreviousText)
+          text << ", ";
+        text << literal.interval_day_to_second().days() << "microseconds";
+      }
+      text << "}_interval_day"; // TODO - Change spec to better name.
+      break;
+    }
+    case ::substrait::proto::Expression_Literal::kFixedChar:
+      // TODO - Should the fixedchar length match its use instead?
```

Comment:
Reworded to "Consider dropping the length in the type in favor of strlen."  Basically the literal string has a length.  Perhaps that's the actual type length instead of something like "fixedchar<21>".  Which means we could just say fixedchar instead.